### PR TITLE
Add migration docs

### PR DESCRIPTION
**Pull request by chaosmail**
Here we go

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+* ~~`tooltip: {mode: 'scrubber'}`~~ can now be styled via CSS
+
+### New version v2
+
+* `dataset: 'dataset00'` defines the dataset key form the new data object in series options
+* ~~`type: 'area'`~~ and ~~`drawDots`~~ in global options is now `type:['area', 'dot']` in series options
+* ~~`lineMode`~~ and ~~`tension`~~ in global options is now `interpolation: {mode: 'cardinal', tension: 0.7}` in series options
+* ~~`grid: true`~~ in series options is now `grid: {x: false, y: true}` in global options
+* ~~`zoomable: true`~~ in series options is now `grid: {x: false, y: true}` in global options
+* ~~`tooltip: {formatter: function(x, y, series) {return 'pouet';}}`~~ in global options is now `tooltipHook` (see [here](http://codepen.io/chaosmail/pen/xZgPmp) for an example usage)
```
Comment (lorem--ipsum) [inline docs/migration.md]: WOuldn't it be better to link to the examples page ?

File: docs/migration.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+* ~~`columnsHGap`~~ is not needed anymore
+* ~~`tooltip: {mode: 'scrubber'}`~~ can now be styled via CSS
+
+### New version v2
+
+* `dataset: 'dataset00'` defines the dataset key form the new data object in series options
+* ~~`type: 'area'`~~ and ~~`drawDots`~~ in global options is now `type:['area', 'dot']` in series options
+* ~~`lineMode`~~ and ~~`tension`~~ in global options is now `interpolation: {mode: 'cardinal', tension: 0.7}` in series options
+* ~~`grid: true`~~ in series options is now `grid: {x: false, y: true}` in global options
+* ~~`zoomable: true`~~ in series options is now `grid: {x: false, y: true}` in global options
```
Comment (lorem--ipsum) [inline docs/migration.md]: `grid` for zoomable ?

**Comment by lorem--ipsum** [comment]
Awesome !

**Comment by chaosmail** [comment]
Alright I updated your comments. The docs on `tooltipHook` are quite difficult to understand if you don't have an example, that's why I also linked the example. Maybe we could just add the relevant code from the example to the docs. Another problem is that due to the SPA using '#' we cannot create links to sections in the docs.

**Comment by lorem--ipsum** [comment]
That makes sense, I'll merge this. Thanks buddy !

**Comment by lorem--ipsum** [comment]
:poultry_leg:

**Pull request merged**